File: conversions/pretty-ms/elide-pretty-ms.py
#!/usr/bin/env python3
"""
Python Integration Example for elide-pretty-ms

This demonstrates calling the TypeScript pretty-ms implementation from Python
using Elide's polyglot capabilities.

Benefits:
- One pretty-ms implementation shared across TypeScript and Python
- Consistent convert milliseconds to human-readable strings across services
- No Python custom implementation needed
- Perfect for performance metrics, duration formatting, UI display
"""

# NOTE: Exact syntax depends on Elide's Python polyglot API (currently alpha)
# This is a CONCEPTUAL example - adjust when Python support is ready

# Assuming Elide provides something like:
# from elide import require
# pretty_ms_module = require('./elide-pretty-ms.ts')

print("=== Python Calling TypeScript pretty-ms ===\n")

# Example: Basic usage
# result = pretty_ms_module.default()
# print(f"Result: {result}")

print("Real-world use case:")
print("- Python service uses convert milliseconds to human-readable strings")
print("- Uses same TypeScript implementation as Node.js API")
print("- Guarantees consistent behavior across entire stack")
print("- Perfect for performance metrics, duration formatting, UI display")
print()

print("Example: Microservices Architecture")
print("┌─────────────────────────────────────┐")
print("│   Elide pretty-ms (TypeScript)         │")
print("│   conversions/pretty-ms/elide-pretty-ms.ts│")
print("└─────────────────────────────────────┘")
print("         ↓                   ↓")
print("    ┌────────┐          ┌────────┐")
print("    │ Node.js│          │ Python │")
print("    │  API   │          │Pipeline│")
print("    └────────┘          └────────┘")
print("         ↓                   ↓")
print("    Same convert milliseconds to human-readable strings everywhere!")
print()

print("Problem Solved:")
print("Before: Python custom implementation + JavaScript = inconsistencies")
print("After: One Elide implementation = 100% consistent")
print()

print("When Elide Python API is ready, usage will be:")
print("  from elide import require")
print("  pretty_ms = require('./elide-pretty-ms.ts')")
print("  # Use it directly!")

File: converted/utilities/pretty-ms/elide-pretty-ms.rb
#!/usr/bin/env ruby

# Ruby Integration Example for elide-pretty-ms
#
# This demonstrates calling the TypeScript pretty-ms implementation from Ruby
# using Elide's polyglot capabilities.
#
# Benefits:
# - One pretty-ms implementation shared across TypeScript and Ruby
# - Consistent convert milliseconds to human-readable strings across microservices
# - Perfect for Rails, Sidekiq, background workers

puts "=== Ruby Calling TypeScript pretty-ms ===\n"

# NOTE: Exact syntax depends on Elide's Ruby polyglot API (currently alpha)
# This is a CONCEPTUAL example - adjust when Ruby support is ready

# Assuming Elide provides something like:
# pretty_ms_module = Elide.require('./elide-pretty-ms.ts')

# Example: Basic usage
# result = pretty_ms_module.default()
# puts "Result: #{result}"

puts "Real-world use case:"
puts "- Ruby worker uses convert milliseconds to human-readable strings"
puts "- Uses same TypeScript implementation as Node.js API"
puts "- Guarantees consistent behavior across services"
puts "- Perfect for performance metrics, duration formatting, UI display"
puts ""

puts "Example: Microservices Architecture"
puts "┌─────────────────────────────────────┐"
puts "│   Elide pretty-ms (TypeScript)         │"
puts "│   conversions/pretty-ms/elide-pretty-ms.ts│"
puts "└─────────────────────────────────────┘"
puts "         ↓           ↓           ↓"
puts "    ┌────────┐  ┌────────┐  ┌────────┐"
puts "    │ Node.js│  │  Ruby  │  │ Rails  │"
puts "    │  API   │  │ Worker │  │  App   │"
puts "    └────────┘  └────────┘  └────────┘"
puts "         ↓           ↓           ↓"
puts "    Same convert milliseconds to human-readable strings everywhere!"
puts ""

puts "Problem Solved:"
puts "Before: Ruby custom + JavaScript = different behaviors"
puts "After: One Elide implementation = 100% consistent"
puts ""

puts "Rails Integration (when ready):"
puts "  # config/initializers/pretty_ms.rb"
puts "  PRETTY_MS = Elide.require('./elide-pretty-ms.ts')"
